### PR TITLE
x86_cpu_flags:test flag disabled

### PR DESCRIPTION
**Pull request by nanliu-r**
ID: 1879026
Signed-off-by: Nana Liu <nanliu@redhat.com>

### CODE REVIEW

**Comment by nanliu-r** [comment]
Test result:
 Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.x86_cpu_flags.others.flag_disable: PASS (190.57 s)

**Comment by nanliu-r** [comment]
@huangyum @vivianQizhu please help to review this, thanks a lot.

Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.9.x86_64.io-github-autotest-qemu.x86_cpu_flags.others.flag_disable: PASS (75.24 s)

File: qemu/tests/x86_cpu_flag_disable.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import logging
+
+from virttest import error_context, env_process
+from qemu.tests.x86_cpu_model import check_flags
```
Comment (vivianQizhu) [inline qemu/tests/x86_cpu_flag_disable.py:4]: The maintainer group now encourage to abstract the common part of multiple cases to tp-qemu/provider/, calling functions from other test cases in tp-qemu is not accepted.

File: qemu/tests/x86_cpu_flag_disable.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    params["start_vm"] = "yes"
+    vm_name = params['main_vm']
+    env_process.preprocess_vm(test, params, env, vm_name)
+
+    vm = env.get_vm(vm_name)
+    cmd = params["check_flag_cmd"]
+    error_context.context("Try to log into guest", logging.info)
+    session = vm.wait_for_login()
+    vm_flags_out = session.cmd_output(cmd).split()
+    if flags[0] in vm_flags_out:
```
Comment (vivianQizhu) [inline qemu/tests/x86_cpu_flag_disable.py:36]: `flags = params["flags"]` It is a string, flags[0] is the first character of this string, is it intended?

File: qemu/tests/x86_cpu_flag_disable.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        test.fail("Flag %s shouldn't in guest" % flags[0])
+
+    check_clock = params.get("check_clock")
+    vm_clock_out = session.cmd_status_output(check_clock)[1]
+    if 'kvmclock' in vm_clock_out:
+        test.fail("kvmclock shouldn't be found")
+
+    if params.get("reboot_method"):
```
Comment (vivianQizhu) [inline qemu/tests/x86_cpu_flag_disable.py:44]: Seems this judgement could be removed, this is no other condition.

File: qemu/tests/x86_cpu_flag_disable.py (hunk 1 above)
Comment (nanliu-r) [reply qemu/tests/x86_cpu_flag_disable.py:4]: Has sent one patch to fix this. [tp-qemu/2409](https://github.com/autotest/tp-qemu/pull/2409)

File: qemu/tests/x86_cpu_flag_disable.py (hunk 2 above)
Comment (nanliu-r) [reply qemu/tests/x86_cpu_flag_disable.py]: You are right, have corrected it and thanks.

File: qemu/tests/x86_cpu_flag_disable.py (hunk 3 above)
Comment (nanliu-r) [reply qemu/tests/x86_cpu_flag_disable.py]: > Seems this judgement could be removed, this is no other condition.

Thanks and update.

File: qemu/tests/x86_cpu_flag_disable.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    2) Otherwise, boot guest with the cpu flags disabled
+    3) Check cpu flags inside guest(only for linux guest)
+    4) Check kvmclock inside guest(only for linux guest)
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    flags = params["flags"]
```
Comment (vivianQizhu) [inline qemu/tests/x86_cpu_flag_disable.py:23]: Name it `flag` would be better, if you want it be a list, flags = params.object('flags'), but here I don't see the necessity.

Comment (nanliu-r) [reply qemu/tests/x86_cpu_flag_disable.py]: > Name it `flag` would be better, if you want it be a list, flags = params.object('flags'), but here I don't see the necessity.

Agree, update, and thanks.

**Comment by vivianQizhu** [comment]
LGTM, let's wait for https://github.com/autotest/tp-qemu/pull/2409

**Comment by vivianQizhu** [comment]
@nanliu-r Resolve conflict please.

**Comment by nanliu-r** [comment]
> @nanliu-r Resolve conflict please.

update，will add the test results later, thanks.

**Comment by nanliu-r** [comment]
(1/2) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.9.x86_64.io-github-autotest-qemu.x86_cpu_flags.others.flag_disable.q35: PASS (46.15 s)
 (2/2) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.x86_cpu_flags.others.flag_disable.q35: PASS (62.24 s)
 (1/1) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.6.10.x86_64.io-github-autotest-qemu.x86_cpu_flags.others.flag_disable: PASS (67.31 s)

**Comment by vivianQizhu** [comment]
@huangyum Request for your ack, thanks.

**Comment by vivianQizhu** [comment]
@huangyum Request for your ack, thanks.

File: qemu/tests/x86_cpu_flag_disable.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    vm = env.get_vm(vm_name)
+    cmd = params["check_flag_cmd"]
+    error_context.context("Try to log into guest", logging.info)
+    session = vm.wait_for_login()
+    vm_flags_out = session.cmd_output(cmd).split()
+    if flag in vm_flags_out:
+        test.fail("Flag %s shouldn't inside guest" % flag)
+
+    check_clock = params.get("check_clock")
+    vm_clock_out = session.cmd_status_output(check_clock)[1]
```
Comment (huangyum) [inline qemu/tests/x86_cpu_flag_disable.py:41]: vm_clock_out = session.cmd_output(check_clock)

File: qemu/tests/x86_cpu_flag_disable.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    vm_name = params['main_vm']
+    env_process.preprocess_vm(test, params, env, vm_name)
+
+    vm = env.get_vm(vm_name)
+    cmd = params["check_flag_cmd"]
+    error_context.context("Try to log into guest", logging.info)
+    session = vm.wait_for_login()
+    vm_flags_out = session.cmd_output(cmd).split()
+    if flag in vm_flags_out:
+        test.fail("Flag %s shouldn't inside guest" % flag)
```
Comment (huangyum) [inline qemu/tests/x86_cpu_flag_disable.py:38]: You can call check_cpu_flags() directly by setting 'no_flags'.

Comment (nanliu-r) [reply qemu/tests/x86_cpu_flag_disable.py:38]: @huangyum  shouldn't we check the flag that shouldn't be in guest?

Comment (huangyum) [reply qemu/tests/x86_cpu_flag_disable.py:38]: Yes, so you need set 'no_flags'.

Comment (nanliu-r) [reply qemu/tests/x86_cpu_flag_disable.py]: Thanks and update

File: qemu/tests/x86_cpu_flag_disable.py (hunk 5 above)
Comment (nanliu-r) [reply qemu/tests/x86_cpu_flag_disable.py]: > vm_clock_out = session.cmd_output(check_clock)

Thanks and update

**Comment by nanliu-r** [comment]
Test pass.
 (1/2) Host_RHEL.m8.u3.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.9.x86_64.io-github-autotest-qemu.x86_cpu_flags.others.flag_disable.q35: PASS (54.43 s)
 (2/2) Host_RHEL.m8.u3.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.x86_cpu_flags.others.flag_disable.q35: PASS (39.68 s)
Host_RHEL.m8.u3.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.6.10.x86_64.io-github-autotest-qemu.x86_cpu_flags.others.flag_disable: PASS (88.68 s)

**Review by huangyum — APPROVED** [review]
LGTM

**Comment by vivianQizhu** [comment]
Thanks @nanliu-r @huangyum

**Pull request merged**